### PR TITLE
fix: x509 domain extraction and leaf certificate extraction

### PR DESCRIPTION
**Pull request by auer-martin**

### CODE REVIEW

File: packages/core/src/utils/domain.ts [hunk 1]
```diff
@@ -1,34 +1,5 @@
...
-    if (starters.indexOf(url[i]) === -1) continue
-    offsetStartSlice = i + 1
-    break
-  }
-
-  if (offsetStartSlice === 0 && offsetPath > 3) return url
-  if (offsetStartSlice > 0 && offsetStartSlice < 2) return ''
-  return url.slice(offsetStartSlice, offsetPath)
+  const regex = /[#/?]/
+  const domain = url.substring('https://'.length).split(regex)[0]
```
Comment (TimoGlastra) [inline packages/core/src/utils/domain.ts:3]: Should we check that it starts with https?

File: packages/core/src/modules/sd-jwt-vc/__tests__/sdjwtvc.fixtures.ts [hunk 2]
```diff
@@ -30,11 +30,18 @@ export const simpleJwtVc =
...
+  trustedCertificate:
+    'MIICdDCCAhugAwIBAgIBAjAKBggqhkjOPQQDAjCBiDELMAkGA1UEBhMCREUxDzANBgNVBAcMBkJlcmxpbjEdMBsGA1UECgwUQnVuZGVzZHJ1Y2tlcmVpIEdtYkgxETAPBgNVBAsMCFQgQ1MgSURFMTYwNAYDVQQDDC1TUFJJTkQgRnVua2UgRVVESSBXYWxsZXQgUHJvdG90eXBlIElzc3VpbmcgQ0EwHhcNMjQwNTMxMDgxMzE3WhcNMjUwNzA1MDgxMzE3WjBsMQswCQYDVQQGEwJERTEdMBsGA1UECgwUQnVuZGVzZHJ1Y2tlcmVpIEdtYkgxCjAIBgNVBAsMAUkxMjAwBgNVBAMMKVNQUklORCBGdW5rZSBFVURJIFdhbGxldCBQcm90b3R5cGUgSXNzdWVyMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEOFBq4YMKg4w5fTifsytwBuJf/7E7VhRPXiNm52S3q1ETIgBdXyDK3kVxGxgeHPivLP3uuMvS6iDEc7qMxmvduKOBkDCBjTAdBgNVHQ4EFgQUiPhCkLErDXPLW2/J0WVeghyw+mIwDAYDVR0TAQH/BAIwADAOBgNVHQ8BAf8EBAMCB4AwLQYDVR0RBCYwJIIiZGVtby5waWQtaXNzdWVyLmJ1bmRlc2RydWNrZXJlaS5kZTAfBgNVHSMEGDAWgBTUVhjAiTjoDliEGMl2Yr+ru8WQvjAKBggqhkjOPQQDAgNHADBEAiAbf5TzkcQzhfWoIoyi1VN7d8I9BsFKm1MWluRph2byGQIgKYkdrNf2xXPjVSbjW/U/5S5vAEC5XxcOanusOBroBbU=',
```
Comment (TimoGlastra) [inline packages/core/src/modules/sd-jwt-vc/__tests__/sdjwtvc.fixtures.ts:42]: Is this the root certificate? Or the issuer certificate?

File: packages/core/src/modules/sd-jwt-vc/__tests__/SdJwtVcService.test.ts [hunk 3]
```diff
@@ -806,15 +808,15 @@ describe('SdJwtVcService', () => {
...
+        isValid: false,
         sdJwtVc: expect.any(Object),
         verification: {
           isSignatureValid: true,
           containsRequiredVcProperties: true,
           containsExpectedKeyBinding: true,
           areRequiredClaimsIncluded: true,
           isValid: true,
-          isValidJwtPayload: true,
+          isValidJwtPayload: false,
```
Comment (TimoGlastra) [inline packages/core/src/modules/sd-jwt-vc/__tests__/SdJwtVcService.test.ts:819]: Why is this false?

File: packages/core/src/utils/domain.ts (hunk 1 above)
Comment (auer-martin) [reply packages/core/src/utils/domain.ts:3]: it already does look a few lines up

File: packages/core/src/modules/sd-jwt-vc/__tests__/sdjwtvc.fixtures.ts (hunk 2 above)
Comment (auer-martin) [reply packages/core/src/modules/sd-jwt-vc/__tests__/sdjwtvc.fixtures.ts:42]: root

File: packages/core/src/modules/sd-jwt-vc/__tests__/SdJwtVcService.test.ts (hunk 3 above)
Comment (auer-martin) [reply packages/core/src/modules/sd-jwt-vc/__tests__/SdJwtVcService.test.ts]: fixed it, forgot to undo it

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**